LongReadRNABenchmark/LongReadRNABenchmarkTasks.wdl

```diff
@@ -0,0 +1,697 @@
+        echo "~{flairBasename} flair" >> gtfs.list
+        echo "~{talonBasename} talon" >> gtfs.list
+
+        /usr/local/src/IsoQuant-3.1.1/misc/denovo_model_stats.py \
+        --gtf_list gtfs.list \
+        --output "~{datasetName}_denovo_stats"
+
+        tar -zcvf ~{datasetName}_denovo_stats.tar.gz ~{datasetName}_denovo_stats
+    >>>
+
+    output {
+        File gffCompareOutput = "~{datasetName}_denovo_stats.tar.gz"
+        File monitoringLog = "monitoring.log"
+    }
+
+    runtime {
+        docker: docker
+        cpu: cpu
+        memory: "~{memory} GiB"
+        disks: "local-disk ~{diskSizeGB} HDD"
+    }
+}
+
+task ReferenceFreeGFFCompare {
+    input {
+        File isoQuantDenovoGTF
+        File stringTieDenovoGTF
+        File expressedGTF
+        String datasetName
+    }
+
+    String docker = "us.gcr.io/broad-dsde-methods/kockan/isoquant-gffcompare:latest"
+    Int cpu = 8
+    Int memory = 64
+    Int diskSizeGB = 300
+    File monitoringScript = "gs://broad-dsde-methods-tbrookin/cromwell_monitoring_script2.sh"
+
+    command <<<
+        bash ~{monitoringScript} > monitoring.log &
+
+        mkdir ~{datasetName}_reference_free_stats
+
+        gffcompare \
+        -r ~{expressedGTF} \
+        -o "~{datasetName}_isoquant_gffcompare" ~{isoQuantDenovoGTF}
+
+        gffcompare \
+        -r ~{expressedGTF} \
+        -o "~{datasetName}_stringtie_gffcompare" ~{stringTieDenovoGTF}
+
+        mv ~{datasetName}* ~{datasetName}_reference_free_stats
+
+        tar -zcvf ~{datasetName}_reference_free_stats.tar.gz ~{datasetName}_reference_free_stats
+    >>>
+
+    output {
+        File gffCompareOutput = "~{datasetName}_reference_free_stats.tar.gz"
+        File monitoringLog = "monitoring.log"
+    }
+
+    runtime {
+        docker: docker
+        cpu: cpu
+        memory: "~{memory} GiB"
+        disks: "local-disk ~{diskSizeGB} HDD"
+    }
+}
+
+task ReducedAnalysisSummarize {
+    input {
+        File reducedGffCompareOutIsoQuant
+        File reducedGffCompareOutStringTie
+        File reducedGffCompareOutBambu
+        File reducedGffCompareOutFlair
+        File reducedGffCompareOutTalon
+        String datasetName
+    }
+
+    String docker = "us.gcr.io/broad-dsde-methods/kockan/kockan-reduced-analysis-summarize:latest"
+    Int cpu = 1
+    Int memory = 32
+    Int diskSizeGB = 50
+
+    command <<<
+        cp ~{reducedGffCompareOutIsoQuant} .
+        cp ~{reducedGffCompareOutStringTie} .
+        cp ~{reducedGffCompareOutBambu} .
+        cp ~{reducedGffCompareOutFlair} .
+        cp ~{reducedGffCompareOutTalon} .
+
+        tar -xzvf ~{datasetName}_isoquant_reduced_db.tar.gz
+        tar -xzvf ~{datasetName}_stringtie_reduced_db.tar.gz
+        tar -xzvf ~{datasetName}_bambu_reduced_db.tar.gz
+        tar -xzvf ~{datasetName}_flair_reduced_db.tar.gz
+        tar -xzvf ~{datasetName}_talon_reduced_db.tar.gz
+
+        python3 /usr/local/src/plot_isoquant_results.py \
```

Comment:
how complicated is this plotting script?  would it work as a heredoc?

Reply:
It's a bit long for my preference. Keeping the WDL clean for now and marking as resolved but can be revisited later.